Failure on Windows-2022 https://github.com/moby/moby/actions/runs/4114098814/jobs/7101259444 in `TestTopWindowsCoreProcesses`. Looks like a flake, but posting here in case it'll happen again. 

```
=== Failed
=== FAIL: github.com/docker/docker/integration-cli TestDockerCLITopSuite/TestTopWindowsCoreProcesses (15.63s)
    check_test.go:359: assertion failed: error is not nil: Error response from daemon: Could not kill running container 24ce24434e62d6a16260e0160300c3e2c6aa6d7e4e080c37f61ae794665e7e83, cannot remove - tried to kill container, but did not receive an exit event: failed to remove 24ce24434e62d6a16260e0160300c3e2c6aa6d7e4e080c37f61ae794665e7e83
    --- FAIL: TestDockerCLITopSuite/TestTopWindowsCoreProcesses (15.63s)

```

--- a/builder/builder-next/worker/worker.go
+++ b/builder/builder-next/worker/worker.go
@@ -18,6 +18,7 @@ import (
 	"github.com/docker/docker/layer"
 	pkgprogress "github.com/docker/docker/pkg/progress"
 	"github.com/moby/buildkit/cache"
+	cacheconfig "github.com/moby/buildkit/cache/config"
 	"github.com/moby/buildkit/client"
 	"github.com/moby/buildkit/client/llb"
 	"github.com/moby/buildkit/executor"
@@ -36,7 +37,6 @@ import (
 	"github.com/moby/buildkit/source/http"
 	"github.com/moby/buildkit/source/local"
 	"github.com/moby/buildkit/util/archutil"
-	"github.com/moby/buildkit/util/compression"
 	"github.com/moby/buildkit/util/contentutil"
 	"github.com/moby/buildkit/util/progress"
 	"github.com/opencontainers/go-digest"
@@ -78,6 +78,10 @@ type Worker struct {
 	Opt
 	SourceManager *source.Manager
 }
+
+var _ interface {
+	GetRemotes(context.Context, cache.ImmutableRef, bool, cacheconfig.RefConfig, bool, session.Group) ([]*solver.Remote, error)
+} = &Worker{}
 
 // NewWorker instantiates a local worker
 func NewWorker(opt Opt) (*Worker, error) {
@@ -227,8 +231,11 @@ func (w *Worker) Exporter(name string, sm *session.Manager) (exporter.Exporter, 
 	}
 }
 
-// GetRemote returns a remote snapshot reference for a local one
-func (w *Worker) GetRemote(ctx context.Context, ref cache.ImmutableRef, createIfNeeded bool, _ compression.Type, s session.Group) (*solver.Remote, error) {
+// GetRemotes returns the remote snapshot references given a local reference
+func (w *Worker) GetRemotes(ctx context.Context, ref cache.ImmutableRef, createIfNeeded bool, _ cacheconfig.RefConfig, all bool, s session.Group) ([]*solver.Remote, error) {
+	if ref == nil {
+		return nil, nil
+	}
 	var diffIDs []layer.DiffID
 	var err error
 	if !createIfNeeded {
@@ -258,10 +265,10 @@ func (w *Worker) GetRemote(ctx context.Context, ref cache.ImmutableRef, createIf
 		}
 	}
 
-	return &solver.Remote{
+	return []*solver.Remote{{
 		Descriptors: descriptors,
 		Provider:    &emptyProvider{},
-	}, nil
+	}}, nil
 }
 
 // PruneCacheMounts removes the current cache snapshots for specified IDs

